### PR TITLE
Add 'Create Wallet' logic in admin

### LINKED ISSUE #8: Setup the Organization wallet
Ref.: VOC-ADM-04

As a requirement for setting up the election on Vocdoni API, as an administrator I need to have an Ethereum Wallet.

As this step can generate friction to Administrators that don't know how to use MetaMask or other wallets, we'll generate a Wallet and save it in the server so we can interact with the Vocdoni API transparently. 

For generating the wallet we'll use the library ether.js. 

- [ ] Given that I'm an administrator
      And there isn't any wallet created in the Organization
      Then I can create a new one
      And it's saved in the server.
- [ ] Given that I'm an administrator
      And there is a wallet created in the Organization
      Then I can't create a new one.

### PR DESCRIPTION
**Pull request by andreslucena**
This PR adds the wallet creation generation and saves it to server so it can be used for operating with the Vocdoni SDK. 

Some notes:

* As this is something that needs to be done once for Organization, it'll not be related to the Steps logic. We only redirect from there if there isn't any wallet created yet.
* You can check that the private key was generated with `bin/rails runner 'puts Decidim::Vocdoni::Wallet.first.private_key'`
* For testing out, you'll need to not have any Wallet created. If you already have one, you can remove it from the command line `bin/rails runner 'Decidim::Vocdoni::Wallet.delete_all'`

Fixes  #8

## Screenshot 

Selection_027

## Testing

- [ ] Given that I'm an administrator
      And there isn't any wallet created in the Organization
      Then I can create a new one
      And it's saved in the server.
- [ ] Given that I'm an administrator
      And there is a wallet created in the Organization
      Then I can't create a new one.

### CODE REVIEW

**Comment by microstudi** [comment]
@andreslucena do you want to rebase this?

**Comment by andreslucena** [comment]
@microstudi I've merged it with main and the specs are passing. This is ready to be reviewed.

File: app/controllers/decidim/vocdoni/admin/wallets_controller.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+        def redirect_location
+          if session[:redirect_back]
+            location = EngineRouter.admin_proxy(current_component).election_steps_path(session[:redirect_back])
+            session[:redirect_back] = nil
+          else
+            location = EngineRouter.admin_proxy(current_component).root_path
+          end
+
+          location
```
Comment (microstudi) [inline app/controllers/decidim/vocdoni/admin/wallets_controller.rb:41]:
```suggestion
          if location = session.delete(:redirect_back)
            EngineRouter.admin_proxy(current_component).election_steps_path(location)
          else
            EngineRouter.admin_proxy(current_component).root_path
          end
```
Not that there's anything wrong but I though you would like this solution

**Review by microstudi — APPROVED** [review]
Everything good with the code, I've left you a suggestion in case you want to apply it.

A part from this, you mention a couple of commands in the description, maybe we could put that information in the Readme.

And, probaly out of scope, has the creation of a wallet always to be random? could we use an existing wallet? I'm just asking this question due my ignorance on how exacly the wallets work on Ethereum

**Comment by andreslucena** [comment]
> Everything good with the code, I've left you a suggestion in case you want to apply it. 

Applied at [f7855e8](https://github.com/decidim-vocdoni/decidim-module-vocdoni/pull/17/commits/f7855e8a10998d0dca426a0dc6adb24661b8fcce) 

> A part from this, you mention a couple of commands in the description, maybe we could put that information in the Readme. 

Ok, I'll add them as a "Troubleshooting/Developing notes" or something like that. It shouldn't be used on production but I guess it's useful for debugging and while coding. I'll do it as a new PR with all the commands once the current set of PRs are merged, as that'd easier to work with git and not having merge conflicts. 

I've add it as a note in the project https://github.com/orgs/decidim-vocdoni/projects/1?query=is%3Aopen+sort%3Aupdated-desc&pane=issue&itemId=19602128 

> And, probaly out of scope, has the creation of a wallet always to be random? could we use an existing wallet? I'm just asking this question due my ignorance on how exacly the wallets work on Ethereum 

First of all, a disclaimer that I'm not an Ethereum expert neither, as I'm still in the process of learning and so on. 

As far as I know, there are different ways of generating Wallets on Ethereum, mostly: 

1. With a browser plugin/mobile application. This is what MetaMask and other providers do. 
2. With a hardware wallet

Then the flow is that the web application asks for permissions, the user accepts it and then the web application has the Wallet's address and can sign transactions on behalf of this user. The problem with this approach on our particular case are two: 

a. It has some usability issues, as you need to explain to the admin that they need to install the add-on, configure it, change it to a Test Net, then go back to Decidim, etc. 
b. As in our case there are multiple admins for an Election (organization admins, spaces aka processes or assemblies admins, etc), then I don't know if the Vocdoni SDK would allow us to work with multiple administrators on their Elections (for instance in the case that we want to implement Pause/Cancel an Election and things like that)

The first problem could be mitigated with documentation and tutorials for the administrators, but is out of the scope of the current MVP. 

The second problem we would need that feature implemented by the SDK. 

On our case, as we're generating the wallet and saving it server-side, both of these problems go away, as the UX is just a button for the admin, and conceptually the admin/creator for Vocdoni is always the same; in fact they call it Organization, just as the model for a Decidim tenant.

**Comment by andreslucena** [comment]
I forget to mention, there's a third way of doing creating the wallet:

3. Using ether.js or other libraries. 

In fact MetaMask recommends using ether.js for the login with Ethereum flow, and I haven't played with hardware wallets but as far as I [see ethers has also some support for that ](https://docs.ethers.org/v5/api/other/hardware/ )

**Review by microstudi — APPROVED** [review]
thanks for the explanations!

**Pull request merged**